### ISSUE TITLE
NPMPlus randomly doesnt start correctly

### ISSUE THREAD

**Issue by Joly0** [ZoeyVid/NPMplus#1317]
i tihnk i already opened an issue for this once or mentioned it somehwere, but here is a dedicated issue for it.

So sometimes, when restarting my server or after updating npmplus, the container does not start correctly, it hangs and doesnt continue starting.
Connecting to the npm webui shows "connection failed" while connecting to any domain that npm manages shows "PR_END_OF_FILE_ERROR" error.

The docker logs for npmplus show this:
```
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/README.md' -> '/data/etc/modsecurity/crs-plugins/README.md'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-after.conf' -> '/data/etc/modsecurity/crs-plugins/empty-after.conf'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-before.conf' -> '/data/etc/modsecurity/crs-plugins/empty-before.conf'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-config.conf' -> '/data/etc/modsecurity/crs-plugins/empty-config.conf'
tls/certbot/config.ini is now unsupported, to remove this warning, just delete the file - some options are replaced by env.
no DEFAULT_CERT_ID set, using dummycerts.
Working on file: /data/nginx/default.conf
Working on file: /data/nginx/proxy_host/1.conf
Working on file: /data/nginx/proxy_host/12.conf
Working on file: /data/nginx/proxy_host/15.conf
Working on file: /data/nginx/proxy_host/16.conf
Working on file: /data/nginx/proxy_host/17.conf
Working on file: /data/nginx/proxy_host/2.conf
Working on file: /data/nginx/proxy_host/21.conf
Working on file: /data/nginx/proxy_host/23.conf
Working on file: /data/nginx/proxy_host/24.conf
Working on file: /data/nginx/proxy_host/25.conf
Working on file: /data/nginx/proxy_host/3.conf
Working on file: /data/nginx/proxy_host/4.conf
Working on file: /data/nginx/proxy_host/5.conf
Working on file: /data/nginx/proxy_host/8.conf
Success.
removed '/usr/local/nginx/logs/nginx.pid'
removed '/run/nginx-1.sock'
removed '/run/nginx-12.sock'
removed '/run/nginx-15.sock'
removed '/run/nginx-16.sock'
removed '/run/nginx-17.sock'
removed '/run/nginx-2.sock'
removed '/run/nginx-21.sock'
removed '/run/nginx-23.sock'
removed '/run/nginx-24.sock'
removed '/run/nginx-25.sock'
removed '/run/nginx-3.sock'
removed '/run/nginx-4.sock'
removed '/run/nginx-5.sock'
removed '/run/nginx-8.sock'
usermod: no changes

-------------------------------------
 _ _  ___  __ __       _
| \ || . \|  \  \ ___ | | _ _  ___
|   ||  _/|     || . \| || | |[_-[
|_\_||_|  |_|_|_||  _/|_| \__|/__/
                 |_|
-------------------------------------
Version:  2.12.1+b091ad0
Date:     Sat Dec 21 13:54:53 CET 2024
User:     npm
PUID:     99
User ID:  99
PGID:     100
Group ID: 100
-------------------------------------


```
thats it, nothing more. Npmplus refuses to continue starting.

Usually restartign npmplus container resolves this issue, it did it this time, but this issue keeps appearing basically everytime there is an update

**Comment by Zoey2936** [comment]
Please run nginx -t inside the container

**Comment by Joly0** [comment]
I will do next time this happens, as i already restarted npmplus and it is working again. I cannot force this behaviour unfortunately, but it usually happens after an update. I´ll still try what i can do.

**Comment by Joly0** [comment]
1a95d371a193:/app# nginx -t
nginx: the configuration file /usr/local/nginx/conf/nginx.conf syntax is
ok
nginx: configuration file /usr/local/nginx/conf/nginx.conf test is successful
1a95d371a193:/app#

**Comment by Zoey2936** [comment]
does the container stops like this or restart?

**Comment by Zoey2936** [comment]
https://github.com/ZoeyVid/NPMplus/blob/develop/rootfs/usr/local/bin/launch.sh because if it sleeps one sleep inf contidions must be true

**Comment by Joly0** [comment]
I am not sure what you mean.

This happend after i restarted my whole server npmplus is running on. Npmplus should have received a term signal on shutdown and shouls have started normally at startup, but it didnt. I am not sure why it should sleep but after restarting the container a second time, it starts as it should

**Comment by Zoey2936** [comment]
npmplus does not start if a php conf or nginxs conf is wrong since this would cause a restart loop

**Comment by Joly0** [comment]
Ok, i understand, but why should a php conf or nginx conf be wrong and how do i check? From the output of the nginx -t command it should be clear, that the nginx conf itself is atleast right, correct?
But why does this only happen after restarting my host or after an update of the container?

**Comment by Joly0** [comment]
This is also not the only system this is happening. I have a vps  beside my home server that runs plain ubuntu instead of unraid and have setup watchtower to automatically update npmplus and i have to constantly manually restart npmplus there too to get it working again

**Comment by Zoey2936** [comment]
maybe the container starts before the files are mounted? and watchtower will not update npmplus

**Comment by Joly0** [comment]
No, the files are 100% certain mounted before npmplus starts. This is also only an issue with npmplus, not any other container and npmplus is not even the first to start.

Also why shouldnt watchtower update npmplus?

**Comment by Zoey2936** [comment]
because watchtower just copies old env will not use new envs => npmplus breaks

**Comment by Joly0** [comment]
What do you mean with "new envs"?
Also watchtower is only used on my vps, not in my home server, so this cant be the reason npmplus is breaking

**Comment by Zoey2936** [comment]
If I add a new env or remove an old env, watchtower will just copy the old ones and changes I did will not apply when updating using watchtower. (Not the reason here) I must say I have no idea what the reason could be since it never happend for me and there are no other reports like this yeet
